src/apps/deskflow-gui/MainWindow.cpp

```diff
@@ -1026,3 +1068,19 @@ void MainWindow::showAndActivate()
   m_actionRestore->setVisible(false);
   m_actionMinimize->setVisible(true);
 }
+
+QString MainWindow::getTlsPath()
+{
+  CoreTool coreTool;
+  return QStringLiteral("%1/%2").arg(coreTool.getProfileDir(), kSslDir);
+}
+
+bool MainWindow::regenerateLocalFingerprints()
+{
+  TlsCertificate tls;
+  if (!tls.generateFingerprint(m_appConfig.tlsCertPath())) {
+    QMessageBox::warning(this, tr("TLS Fingerprint Error"), tr("Failed to Cacluate new Keys"));
```

Comment:
Typo ("Cacluate"). Have you tried [cspell](https://cspell.org/)?

Also, only proper nouns need capitalizing in sentence case.

```suggestion
    QMessageBox::warning(this, tr("TLS Fingerprint Error"), tr("Failed to calculate new keys"));
```

![image](https://github.com/user-attachments/assets/99a2f434-221a-4ab8-a9f3-dba0da338ef0)

Or, we could use `qCritical()` which in our app shows a helpful diagnostics dialog with the version number, link to report bug, etc. It also logs the error which is helpful when understanding end-user issues.

```suggestion
    qCritical() << "failed to calculate new keys";
```
![image](https://github.com/user-attachments/assets/5e73a737-b292-4197-8403-b5c8c9a40b90)